recipe-client-addon/lib/RecipeRunner.jsm

```diff
@@ -23,22 +24,33 @@ this.EXPORTED_SYMBOLS = ["RecipeRunner"];
 
 const log = LogManager.getLogger("recipe-runner");
 const prefs = Services.prefs.getBranch("extensions.shield-recipe-client.");
+const TIMER_NAME = "recipe-client-addon-run";
 
 this.RecipeRunner = {
   init() {
     if (!this.checkPrefs()) {
       return;
     }
 
-    let delay;
     if (prefs.getBoolPref("dev_mode")) {
-      delay = 0;
-    } else {
-      // startup delay is in seconds
-      delay = prefs.getIntPref("startup_delay_seconds") * 1000;
+      // Run right now in dev mode
+      this.run();
     }
 
-    setTimeout(this.start.bind(this), delay);
+    // Run once every `runInterval` wall-clock seconds.
+    // This is managed by setting a "last ran" timestamp, and running if it is
+    // more than `runInterval` seconds ago.
```

Comment:
It is hard to put succintly, but there [are](https://dxr.mozilla.org/mozilla-central/source/toolkit/components/timermanager/nsUpdateTimerManager.js#84-87) [several](https://dxr.mozilla.org/mozilla-central/source/toolkit/components/timermanager/nsUpdateTimerManager.js#128-130) [reasons](https://dxr.mozilla.org/mozilla-central/source/toolkit/components/timermanager/nsUpdateTimerManager.js#164-169) why the timer might not run exactly on time, or on very short intervals. I'd expect timers of about 5 minutes to work well, and any shorter would start to exhibit weird behavior.